core/src/main/java/org/apache/accumulo/core/iterators/YieldCallback.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators;
+
+/**
+ * This callback handles the state of yielding within an iterator
+ */
+public class YieldCallback<K> {
```

Comment:
Transform could possibly be handled in a push down manner also.  This could be made easier with a utility method.  Just sharing thoughts I had on this for discussion, not saying this is something we should do.

```java
class YieldCallback {
  /**
   * Returns a new YieldCallback that wraps the passed in a transforms the key when yeild is called.
   */
  YieldCallback transform(YieldCallback yc, Function<Key, Key> keyTransform);
}
```

```java
class MyIter implements SKVI {
  @Override
  void enableYielding(YieldCallback yc) {
    source.enableYielding(YieldCallback.transform(yc, k -> transform(k));
  }
}

```